homeassistant/components/deconz/binary_sensor.py

```diff
@@ -79,7 +79,9 @@ class DeconzBinarySensorDescription(
     CarbonMonoxide: [
         DeconzBinarySensorDescription(
             key="carbon_monoxide",
-            value_fn=lambda device: device.carbon_monoxide,  # type: ignore[no-any-return]
+            value_fn=lambda device: device.carbon_monoxide
+            if isinstance(device, CarbonMonoxide)
+            else None,
```

Comment:
Since you asked 😄 
I thought about maybe adding an helper function for this which will help shorten this back into 1 line and do the check for the instance, something like:
```python
def _get_value(device, instance, value):
  if isinstance(device, instance):
    return value
  return None
 ```